### PR TITLE
Add image-text-to-text and edit image-to-text task pages

### PR DESCRIPTION
**Pull request by merveenoyan**

### CODE REVIEW

File: packages/tasks/src/tasks/image-text-to-text/data.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import type { TaskDataCustom } from "..";
+
+const taskData: TaskDataCustom = {
+	datasets: [
+		{
+			// TODO write proper description
```
Comment (osanseviero) [inline packages/tasks/src/tasks/image-text-to-text/data.ts:6]: Should we include https://huggingface.co/datasets/HuggingFaceM4/WebSight ?

File: packages/tasks/src/tasks/image-text-to-text/data.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+			description: "Instructions composed of image and text.",
+			id: "liuhaotian/LLaVA-Instruct-150K",
+		},
+		{
+			// TODO write proper description
```
Comment (osanseviero) [inline packages/tasks/src/tasks/image-text-to-text/data.ts:11]: TODO

File: packages/tasks/src/tasks/image-text-to-text/data.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        {
+			description: "A strong document understanding model.",
+			id: "microsoft/udop-large",
+		},
+		{
+			description: "A powerful model that lets you have a conversation with the image.",
+			id: "llava-hf/llava-1.5-7b-hf",
+		},
+	],
+	spaces: [
```
Comment (osanseviero) [inline packages/tasks/src/tasks/image-text-to-text/data.ts:55]: Is there a nice llava or idefics demo we can put here?

File: packages/tasks/src/tasks/image-text-to-text/data.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+			description: "An application to compare variants of a document vision language model.",
+			id: "merve/pix2struct",
+		},
+        {
+			description: "An application to compare outputs of different vision language models.",
+			id: "merve/compare_VLMs",
+		},
+	],
+	summary:
+		"Image-text-to-text models take in an image and text prompt and output text. These models are also called vision language models. The difference from image-to-text models is that these models take an additional text input, not restricting model to certain use cases like image captioning.",
```
Comment (osanseviero) [inline packages/tasks/src/tasks/image-text-to-text/data.ts:74]: They can even take as input a whole conversation, they are not limited to one prompt - maybe we should mention that. wdyt?

File: packages/tasks/src/tasks/image-text-to-text/about.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
+## Use Cases
+
+### Visual Question Answering
```
Comment (osanseviero) [inline packages/tasks/src/tasks/image-text-to-text/about.md:3]: I would avoid to put examples of things that are covered in other task pages to avoid confusion. The 3 examples now are already covered. E.g.

- VQA https://huggingface.co/tasks/visual-question-answering
- Doc QA https://huggingface.co/tasks/document-question-answering
- Captioning https://huggingface.co/tasks/image-to-text

File: packages/tasks/src/tasks/image-text-to-text/about.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+Vision language models trained on document text, layout and graphics can be used to extract information from documents.
+
+### Image Captioning
+
+Vision language models can be used to generate captions for images.
+
+## Inference
+
+You can use the 🤗 Transformers library's `image-text-to-text` pipeline to interact with vision language models.
```
Comment (osanseviero) [inline packages/tasks/src/tasks/image-text-to-text/about.md:17]:
```suggestion
You can use the Transformers library's `image-text-to-text` pipeline to interact with vision language models.
```

**Review by osanseviero — CHANGES_REQUESTED** [review]
Looking nice!

File: packages/tasks/src/tasks/image-text-to-text/data.ts (hunk 4 above)
Comment (merveenoyan) [reply packages/tasks/src/tasks/image-text-to-text/data.ts:74]: Yes indeed.

File: packages/tasks/src/tasks/image-text-to-text/about.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+Vision language models trained on document text, layout and graphics can be used to extract information from documents.
+
+### Image Captioning
+
+Vision language models can be used to generate captions for images.
+
+## Inference
+
+You can use the Transformers library's `image-text-to-text` pipeline to interact with vision language models.
```
Comment (pcuenca) [inline packages/tasks/src/tasks/image-text-to-text/about.md:17]:
```suggestion
You can use the Transformers library's `image-text-to-text` pipeline to interact with vision-language models.
```

File: packages/tasks/src/tasks/image-text-to-text/data.ts [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+	models: [
+		{
+			description: "A powerful and grounded vision language model that can also localize concepts in images.",
+			id: "microsoft/kosmos-2-patch14-224",
+		},
+		{
+			description: "A strong vision language model that can also localize texts in images.",
+			id: "adept/fuyu-8b",
+		},
+        {
```
Comment (pcuenca) [inline packages/tasks/src/tasks/image-text-to-text/data.ts:46]:
```suggestion
		{
```

File: packages/tasks/src/tasks/index.ts [hunk 9]
```diff
@@ -190,6 +191,7 @@ export const TASKS_DATA: Record<PipelineType, TaskData | undefined> = {
 	"image-segmentation": getData("image-segmentation", imageSegmentation),
 	"image-text-to-text": undefined,
 	"image-to-image": getData("image-to-image", imageToImage),
+	"image-text-to-text": getData("image-text-to-text", imageTextToText),
```
Comment (pcuenca) [inline packages/tasks/src/tasks/index.ts:194]:
```suggestion
	"image-to-image": getData("image-to-image", imageToImage),
	"image-text-to-text": getData("image-text-to-text", imageTextToText),
```

Duplicate entry above

**Review by pcuenca — COMMENTED** [review]
Fixed a couple of formatting issues. Agree with Omar's comments :)

File: packages/tasks/src/tasks/image-text-to-text/about.md (hunk 5 above)
Comment (merveenoyan) [reply packages/tasks/src/tasks/image-text-to-text/about.md:3]: I think I disagree here. one could segment humans with both image segmentation and mask detection for instance (except for zero shot part). some hugging face tasks are similar. some models are very generalistic that they can handle these tasks altogether and the paradigm is shifting more towards there anyway.

these models are similar either way, if I were an MLE who had to do captioning I'd like to try both VLMs (BTW which perform better IMO) and direct captioning models (image-to-text) which has sole purpose of captioning. I wouldn't like to keep this information from the user.

File: packages/tasks/src/tasks/image-text-to-text/data.ts (hunk 8 above)
Comment (merveenoyan) [reply packages/tasks/src/tasks/image-text-to-text/data.ts:46]: I'll format locally once the content is approved, thanks though

File: packages/tasks/src/tasks/image-text-to-text/data.ts [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+			// TODO write proper description
+			description: "Instructions composed of image and text.",
+			id: "liuhaotian/LLaVA-Instruct-150K",
+		},
+		{
+			description: "Conversation turns where questions involve image and text.",
+			id: "liuhaotian/LLaVA-Pretrain",
+		},
+		{
+			description: "Screenshots of websites with their codes.",
```
Comment (pcuenca) [inline packages/tasks/src/tasks/image-text-to-text/data.ts:15]:
```suggestion
			description: "Screenshots of websites with their HTML/CSS codes.",
```

File: packages/tasks/src/tasks/image-text-to-text/data.ts [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+			description: "Powerful vision-language model assistant.",
+			id: "badayvedat/LLaVA",
+		},
+		{
+			description: "An application to compare outputs of different vision language models.",
+			id: "merve/compare_VLMs",
+		},
+	],
+	summary:
+		"Image-text-to-text models take in an image and text prompt and output text. These models are also called vision language models. The difference from image-to-text models is that these models take an additional text input, not restricting model to certain use cases like image captioning, and can take a conversation as input.",
```
Comment (pcuenca) [inline packages/tasks/src/tasks/image-text-to-text/data.ts:82]:
```suggestion
		"Image-text-to-text models take in an image and text prompt and output text. These models are also called vision-language models. The difference from image-to-text models is that these models take an additional text input, not restricting the model to certain use cases like image captioning, and may also be trained to accept a conversation as input.",
```

File: packages/tasks/src/tasks/image-to-text/about.md [hunk 12]
```diff
@@ -22,24 +18,11 @@ You can use the 🤗 Transformers library's `image-to-text` pipeline to generate
 ```python
 from transformers import pipeline
 
-captioner = pipeline("image-to-text",model="Salesforce/blip-image-captioning-base")
+captioner = pipeline("image-to-text", model="Salesforce/blip-image-captioning-base")
```
Comment (pcuenca) [inline packages/tasks/src/tasks/image-to-text/about.md:21]: 👍

File: packages/tasks/src/tasks/image-text-to-text/about.md (hunk 5 above)
Comment (merveenoyan) [reply packages/tasks/src/tasks/image-text-to-text/about.md:3]: I think it would be too repetitive. See here for instance the one in data.ts will be at the top, and then right after use case one repeats it. I wouldn't like to add it there
Screenshot 2024-03-25 at 20 49 41

Comment (osanseviero) [reply packages/tasks/src/tasks/image-text-to-text/about.md:3]: My main concern is that now we have two tasks that cover the same thing, so it could end up confusing users. As an example, imagine if https://huggingface.co/tasks/text-generation use cases were Translation and Summarization, which are also their own separate tasks.

I wonder if we can use here specific applications rather than sub-tasks covered here (thinking of https://huggingface.co/tasks/text-generation#use-cases as a nice example)

**Comment by merveenoyan** [comment]
fyi @pcuenca this PR doesn't have to be merged until Niels' pipeline PR is merged imo

File: packages/tasks/src/tasks/image-text-to-text/about.md (hunk 5 above)
Comment (merveenoyan) [reply packages/tasks/src/tasks/image-text-to-text/about.md:3]: yes I agree, but I think as a specific/solid use case there's not a lot (they'd be way too specific)

Comment (merveenoyan) [reply packages/tasks/src/tasks/image-text-to-text/about.md:3]: an organic way for me to infer use cases for tasks were checking out example inputs to many Spaces based on a specific task, and for VLMs it's mostly visual question answering for instance.

Comment (merveenoyan) [reply packages/tasks/src/tasks/image-text-to-text/about.md:15]: I added bunch of other things

**Comment by merveenoyan** [comment]
I think the pipeline PR in transformers is nowhere near to be merged. Would be great if you could review again @osanseviero @pcuenca we could merge this week and still launch

File: packages/tasks/src/tasks/image-text-to-text/data.ts [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+			{
+				label: "Answer",
+				content:
+					"The bee is sitting on a pink flower, surrounded by other flowers. The bee is positioned in the center of the flower, with its head and front legs sticking out.",
+				type: "text",
+			},
+		],
+	},
+	metrics: [],
+	models: [
```
Comment (osanseviero) [inline packages/tasks/src/tasks/image-text-to-text/data.ts:40]: moondream and paligemma might be interesting (replacing some of the current older ones)

File: packages/tasks/src/tasks/image-text-to-text/data.ts [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+		{
+			description: "A strong document understanding model.",
+			id: "microsoft/udop-large",
+		},
+		{
+			description: "A powerful model that lets you have a conversation with the image.",
+			id: "llava-hf/llava-1.5-7b-hf",
+		},
+	],
+	spaces: [
```
Comment (osanseviero) [inline packages/tasks/src/tasks/image-text-to-text/data.ts:58]: to update - we had lots of nice ones recently I think

File: packages/tasks/src/tasks/image-text-to-text/about.md [hunk 15]
```diff
@@ -0,0 +1,43 @@
+## Use Cases
+
+### Multimodal Dialogue
+
+Vision language models can be used as multimodal assistants, keeping context about the conversation and keeping the image to have multiple turn dialogues.
```
Comment (osanseviero) [inline packages/tasks/src/tasks/image-text-to-text/about.md:5]:
```suggestion
Vision language models can be used as multimodal assistants, keeping context about the conversation and keeping the image to have multiple-turn dialogues.
```

File: packages/tasks/src/tasks/image-text-to-text/about.md [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+
+### Image Recognition with Instructions
+
+Vision language models can recognize images through descriptions. When given detailed descriptions of specific entities, it can classify the entities in an image.
+
+### Zero-shot Object Detection, Image Segmentation and Localization
+
+Some vision language models can detect or segment a set of objects or describe the positions or relative positions of the objects. For example, one could prompt such a model to ask if one object is behind another. Such a model can also output bounding box coordination or segmentation masks directly in the text output, unlike the traditional models explicitly trained on only doing object detection or image segmentation.
```
Comment (osanseviero) [inline packages/tasks/src/tasks/image-text-to-text/about.md:13]:
```suggestion
Some vision language models can detect or segment a set of objects or describe the positions or relative positions of the objects. For example, one could prompt such a model to ask if one object is behind another. Such a model can also output bounding box coordination or segmentation masks directly in the text output, unlike the traditional models explicitly trained on only object detection or image segmentation.
```

File: packages/tasks/src/tasks/image-text-to-text/about.md [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+
+Vision language models trained on image-text pairs can be used for visual question answering.
+
+### Document Question Answering and Retrieval
+
+Vision language models trained on formatted documents that include text and graphics can extract information from documents. This includes OCR-free retrieval as well, where one can directly feed an image without putting the image through OCR and feeding the text.
+
+### Image Captioning
+
+Vision language models can be used to generate captions for images.
```
Comment (osanseviero) [inline packages/tasks/src/tasks/image-text-to-text/about.md:25]: I feel this is a bit redundant/repetitive with the heading. Maybe everything can be encompassed under zero-shot capabilities, and have a bullet point list of some example tasks

File: packages/tasks/src/tasks/image-text-to-text/about.md [hunk 18]
```diff
@@ -0,0 +1,43 @@
...
+from transformers import pipeline
+
+mm_pipeline = pipeline("image-text-to-text", model="microsoft/kosmos-2-patch14-224")
+
+generated_text = mm_pipeline(images="https://huggingface.co/spaces/llava-hf/llava-4bit/resolve/main/examples/baklava.png", text="How to make this pastry?", max_new_tokens=50)
+## [{'generated_text': 'How to make this pastry? 1. Preheat oven to 200 degrees Celsius.'}]
+```
+
+## Useful Resources
+
```
Comment (osanseviero) [inline packages/tasks/src/tasks/image-text-to-text/about.md:41]: Maybe https://huggingface.co/blog/idefics too

**Comment by merveenoyan** [comment]
@osanseviero can you review again?

File: packages/tasks/src/tasks/image-text-to-text/about.md [hunk 19]
```diff
@@ -0,0 +1,75 @@
+## Different Types of Vision Language Models
+
+Vision language models come in three types:
+
+- **Base:** Pre-trained models that can be fine-tuned. A good example is for base models is [PaliGemma models](https://huggingface.co/models?sort=trending&search=google%2Fpaligemma-3b-pt) by Google.
```
Comment (osanseviero) [inline packages/tasks/src/tasks/image-text-to-text/about.md:5]:
```suggestion
- **Base:** Pre-trained models that can be fine-tuned. A good example of base models is the [PaliGemma models family](https://huggingface.co/models?sort=trending&search=google%2Fpaligemma-3b-pt) by Google.
```

File: packages/tasks/src/tasks/image-text-to-text/about.md [hunk 20]
```diff
@@ -0,0 +1,75 @@
...
+- **Instruction:** Base models fine-tuned on instruction datasets. A good example for instruction fine-tuned models is [idefics2-8b](https://huggingface.co/HuggingFaceM4/idefics2-8b).
```
Comment (osanseviero) [inline packages/tasks/src/tasks/image-text-to-text/about.md:6]:
```suggestion
- **Instruction:** Base models fine-tuned on instruction datasets. A good example of instruction fine-tuned models is [idefics2-8b](https://huggingface.co/HuggingFaceM4/idefics2-8b).
```

File: packages/tasks/src/tasks/image-text-to-text/about.md [hunk 21]
```diff
@@ -0,0 +1,75 @@
...
+
+Some vision language models can detect or segment a set of objects or describe the positions or relative positions of the objects. For example, one could prompt such a model to ask if one object is behind another. Such a model can also output bounding box coordination or segmentation masks directly in the text output, unlike the traditional models explicitly trained on only object detection or image segmentation.
+
+### Visual Question Answering
+
+Vision language models trained on image-text pairs can be used for visual question answering and generating captions for images.
+
+### Document Question Answering and Retrieval
+
+Documents often consist of different layouts, charts, tables, images and more. Vision language models trained on formatted documents can extract information from documents. This is an OCR-free approach, the inputs skip OCR and documents are directly fed to vision language models.
```
Comment (osanseviero) [inline packages/tasks/src/tasks/image-text-to-text/about.md:27]:
```suggestion
Documents often consist of different layouts, charts, tables, images, and more. Vision language models trained on formatted documents can extract information from them. This is an OCR-free approach; the inputs skip OCR, and documents are directly fed to vision language models.
```

File: packages/tasks/src/tasks/image-text-to-text/about.md [hunk 22]
```diff
@@ -0,0 +1,75 @@
...
+```python
+from transformers import LlavaNextProcessor, LlavaNextForConditionalGeneration
+import torch
+
+device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+processor = LlavaNextProcessor.from_pretrained("llava-hf/llava-v1.6-mistral-7b-hf")
+model = LlavaNextForConditionalGeneration.from_pretrained(
+    "llava-hf/llava-v1.6-mistral-7b-hf",
+    torch_dtype=torch.float16,
+    low_cpu_mem_usage=True
```
Comment (osanseviero) [inline packages/tasks/src/tasks/image-text-to-text/about.md:46]: I don't think we should put such a feature (somewhat experimental) in the task pages

**Review by osanseviero — APPROVED** [review]
LGTM with some minor comments, thanks!
